frontend/src/metabase/containers/SaveQuestionModal/context.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  createContext,
+  type PropsWithChildren,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from "react";
+
+import { useListCollectionsQuery } from "metabase/api";
+import { SAVE_QUESTION_SCHEMA } from "metabase/containers/SaveQuestionModal/schema";
+import type {
+  FormValues,
+  SaveQuestionProps,
+} from "metabase/containers/SaveQuestionModal/types";
+import {
+  getInitialValues,
+  getPlaceholder,
+  getTitle,
+  submitQuestion,
+} from "metabase/containers/SaveQuestionModal/util";
+import { FormProvider } from "metabase/forms";
+import { useSelector } from "metabase/lib/redux";
+import { getIsSavedQuestionChanged } from "metabase/query_builder/selectors";
+import type Question from "metabase-lib/v1/Question";
+
+type SaveQuestionContextType = {
+  question: Question;
+  originalQuestion: Question | null;
+  title: string;
+  nameInputPlaceholder: string;
+  initialValues: FormValues;
+  handleSubmit: (details: FormValues) => Promise<void>;
+  values: FormValues;
+  setValues: (values: FormValues) => void;
+  showSaveType: boolean;
+};
+
+export const SaveQuestionContext =
+  createContext<SaveQuestionContextType | null>(null);
+
+export const SaveQuestionProvider = ({
+  question,
+  originalQuestion: latestOriginalQuestion,
+  onCreate,
+  onSave,
+  multiStep,
+  initialCollectionId,
+  children,
+}: PropsWithChildren<SaveQuestionProps>) => {
+  const { data: collections = [] } = useListCollectionsQuery({});
+  const [originalQuestion] = useState(latestOriginalQuestion); // originalQuestion from props changes during saving
```

Comment:
I haven't used this trick before, usually I would use `useRef` but if this also works then it should be okay

Reply:
yeah, this isn't my code - it's been there for a while now. I'd rather leave it as-is since it's been working for years